### PR TITLE
Add row and address component

### PR DESCRIPTION
**Pull request by FrederikBolding**
Adds a new row UI component for improving the layout of custom UIs alongside an address component that can be used inside this row.

Also updates the transaction insight example snap to use these new components and moves some code to `snaps-sdk` to support validation  of the components.

Progresses https://github.com/MetaMask/MetaMask-planning/issues/1570

### CODE REVIEW

File: packages/snaps-sdk/src/ui/components/row.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * Create a {@link Row} node.
+ *
+ * @param args - The node arguments. This can either be a string, a component and an optional variant or an object
+ * with the properties: `label`, `value` and `variant`.
+ * @param args.label - The label for the row.
+ * @param args.value - Another component, is currently limited to `image`, `text` and `address`.
+ * @param args.variant - An optional variant, either `default`, `warning` or `critical`.
+ * @returns The row node as an object.
+ * @example
+ * const node = image({ value: '<svg />' });
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/ui/components/row.ts:57]:
```suggestion
```

File: packages/snaps-sdk/jest.config.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ module.exports = deepmerge(baseConfig, {
       branches: 100,
       functions: 100,
       lines: 100,
-      statements: 100,
+      statements: 99.55,
```
[the remark this answers is not in the document]
Comment (FrederikBolding) [reply packages/snaps-sdk/jest.config.js]: Fixed!

File: packages/examples/packages/transaction-insights/src/index.ts [hunk 3]
```diff
@@ -30,10 +30,17 @@ export const onTransaction: OnTransactionHandler = async ({ transaction }) => {
   ) {
     const type = decodeData(transaction.data);
     return {
-      content: panel([text('**Transaction type:**'), text(type)]),
+      content: panel([
+        row('From Address', address(transaction.from)),
+        row(
+          'To Address',
+          transaction.to ? address(transaction.to) : text('None'),
+        ),
```
Comment (Mrtenz) [inline packages/examples/packages/transaction-insights/src/index.ts:38]: Isn't it implied by the `address` component that these are addresses?

```suggestion
        row('From', address(transaction.from)),
        row(
          'To',
          transaction.to ? address(transaction.to) : text('None'),
        ),
```

Comment (FrederikBolding) [reply packages/examples/packages/transaction-insights/src/index.ts]: yeah!

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**